src/Workspaces/CoreTest/EditorConfigStorageLocation/NamingStylePreferenceEditorConfigStorageLocationTests.cs

```diff
@@ -42,38 +33,159 @@ public static void TestEmptyDictionaryDefaultNamingStylePreferencesObjectReturns
+            Assert.True(result, "Expected non-empty dictionary to return true");
+            var isNamingStylePreferencesObject = combinedNamingStyles is NamingStylePreferences;
+            Assert.True(isNamingStylePreferencesObject, $"Expected returned object to be of type '{nameof(NamingStylePreferences)}'");
+            Assert.Equal($"{firstNameAfterOrdering}_style", ((NamingStylePreferences)combinedNamingStyles).Rules.NamingRules[0].NamingStyle.Name);
+            Assert.Equal($"{secondNameAfterOrdering}_style", ((NamingStylePreferences)combinedNamingStyles).Rules.NamingRules[1].NamingStyle.Name);
+        }
+
+        [Theory]
+        [InlineData("a", "b", "a", "method", "method, property")]
+        [InlineData("b", "a", "a", "method, property", "method")]
+        [InlineData("b", "a", "b", "method", "method, property")]
+        [InlineData("a", "b", "b", "method, property", "method")]
+        [InlineData("a", "b", "a", "*", "*")]
+        [InlineData("b", "a", "a", "*", "*")]
+        [InlineData("A", "b", "A", "*", "*")]
+        [InlineData("b", "A", "A", "*", "*")]
+        [InlineData("a", "B", "a", "*", "*")]
+        [InlineData("B", "a", "a", "*", "*")]
+        [InlineData("A", "B", "A", "*", "*")]
+        [InlineData("B", "A", "A", "*", "*")]
+        public static void TestOrderedBySymbolsBeforeName(string firstName, string secondName, string firstNameAfterOrdering, string firstSymbols, string secondSymbols)
+        {
+            var editorConfigStorageLocation = new NamingStylePreferenceEditorConfigStorageLocation();
+            var newDictionary = new Dictionary<string, string>()
+            {
+                [$"dotnet_naming_rule.{firstName}.severity"] = "error",
+                [$"dotnet_naming_rule.{firstName}.symbols"] = "first_symbols",
+                [$"dotnet_naming_rule.{firstName}.style"] = $"{firstName}_style",
+                ["dotnet_naming_symbols.first_symbols.applicable_kinds"] = firstSymbols,
+                [$"dotnet_naming_style.{firstName}_style.capitalization"] = "pascal_case",
+                [$"dotnet_naming_style.{secondName}_style.capitalization"] = "camel_case",
+                [$"dotnet_naming_rule.{secondName}.severity"] = "error",
+                [$"dotnet_naming_rule.{secondName}.symbols"] = "second_symbols",
+                [$"dotnet_naming_rule.{secondName}.style"] = $"{secondName}_style",
+                ["dotnet_naming_symbols.second_symbols.applicable_kinds"] = secondSymbols,
+            };
+
+            var result = editorConfigStorageLocation.TryGetOption(
+                newDictionary,
+                typeof(NamingStylePreferences),
+                out var combinedNamingStyles);
+
+            var secondNameAfterOrdering = firstNameAfterOrdering == firstName ? secondName : firstName;
+            Assert.True(result, "Expected non-empty dictionary to return true");
+            var isNamingStylePreferencesObject = combinedNamingStyles is NamingStylePreferences;
```

Comment:
See earlier comment about type assertions.

Reply:
➡️ Covering this with https://github.com/dotnet/roslyn/pull/36366#discussion_r293152657